buildLib/commandLineHelper.mjs

```diff
@@ -0,0 +1,234 @@
+import process from 'node:process';
+
+/**
+ * @file This file handles parsing command line arguments for any custom build tools used in building opentype.js
+ */
+
+export default function createCommandLineArgumentsHandler(validArguments) {
```

Comment:
if I understand correctly, SWC use JSON as argument format,
and since there is no official CLI wrapper, we do the --arg => JSON conversion with this mjs file

If that's the case, my question is : could we store those arguments (as JSON) in our package.json (not far from the the builds steps) so we could have a more direct control on SWC (without having to be limited by this intermediate CLI values representation)

Reply:
> if I understand correctly, SWC use JSON as argument format, and since there is no official CLI wrapper, we do the --arg => JSON conversion with this mjs file
> 
> If that's the case, my question is : could we store those arguments (as JSON) in our package.json (not far from the the builds steps) so we could have a more direct control on SWC (without having to be limited by this intermediate CLI values representation)

No, this is not an arguments to json converter for SWC. This validates arguments passed to esbuild-runner.mjs to confirm they're valid (using a config passed to it from esbuild-runner.mjs) and generates a json object representing them that is more easily used from the code in esbuild-runner.mjs.

The purpose of esbuild-runner is to generate the correct SWC and ESBuild configs for the target specified.

I did move the configuration for the compat target to the package.json